### ISSUE TITLE
[1.16] Normalisation in Kriging algorithm 

### ISSUE THREAD

**Issue by KieranDelamotte** [openturns/openturns#1667]
A small question / remark which may be useful for 1.16 users. 

The data are not normalized anymore in the Kriging algorithm and users must do it on their own. How can we place this information ? I already use the `IsoProbabilisticTransformation ` method in the examples but there's maybe a better way to do it ?

Kieran

**Comment by jschueller** [comment]
I guess it depends on the context, as you known your distribution IsoProbabilisticTransformation is a good way, but when the distribution is unknown you have to rely on a custom normalization function.

**Comment by josephmure** [comment]
The [Kriging the cantilever beam](http://openturns.github.io/openturns/1.16/auto_meta_modeling/kriging_metamodel/plot_kriging_cantilever_beam.html) example is currently broken in the 1.16 doc because the data are not transformed. Is `IsoProbabilisticTransformation` really the answer to this? I think there are 2 reasonable solutions:

1. Initiate the solver at a more sensible starting point.
2. Apply a linear transform to the data and then apply the reverse transform to the value of the scale parameter found by the solver.

**Comment by KieranDelamotte** [comment]
@jschueller  Thanks for your input.

@josephmure The link you mention seems OK now isn't it ?

Also I agree the `IsoProbabilisticTransformation` is a bit specific and I can use it since I know perfectly the model. My question was more from a user point of view. I no normalization is done, which is OK, then the information should be somewhere. As for the best practice I guess a simple linear transform should be enough. Is the Box-Cox transform an adapted tool for this ?

**Comment by josephmure** [comment]
Yes the link works now! When was the doc updated? I could have sworn it was still broken when I wrote the comment  O_o

**Comment by sofianehaddad** [comment]
With #1530 and #1572 I used both `IsoProbabilisitcTransformation` and more custom `transformations` (linear ones using data) to illustrate the need to normalize data for this `KrigingAlgorithm` even if it is not the best place.  @KieranDelamotte I think `Box-Cox` might be adapted but basic transformations (linear) might do the job.

I think we should add a huge warn in `KrigingAlgorithm` & `GeneralLinearModelAlgorithm` API doc about this new feature/regression.

For `1.17` I think we need basic transformations in the spirit of [this preprocessing](https://scikit-learn.org/stable/modules/preprocessing.html) module (see [API contents](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.preprocessing) for more details). It is a discussion we had once during a CT and think today we definitely should go for with at least a `StandardTransformation`, `CDFTransformation`

**Comment by mbaudin47** [comment]
I created a thread to discuss this:

https://openturns.discourse.group/t/normalization-of-the-input-sample-in-krigingalgorithm-in-ot1-16/101

Actually, the `IsoProbabilisitcTransformation` cannot be the way to go in general in my opinion, because this may involve a complex nonlinear transformation if a general copula is used. The Box-Cox transformation is not specifically design for this, so I guess that we might easily find a counter-example. Please consider the text in the forum as a rationale behind the current choice. 

I think that there are two way to go:
* do as if the input had uniform distribution and scale by (x - xmin) / (xmax - xmin),
* do as if the input had gaussian distribution and scale by (x - mu) / sigma,

The current option uses the "uniform" scaling, which is consistent that the scaling parameter must be given bounds to run the optimization algorithm safely. The second scaling may perform better if the sample size is to small to estimate the xmin and xmax parameter sufficiently accurately. Whatever the choice we make, i guess that providing a scaling feature with different options would be great for users. 

Also notice that scikit-learn has a whole collection of normalization methods:

https://scikit-learn.org/stable/auto_examples/preprocessing/plot_all_scaling.html#sphx-glr-auto-examples-preprocessing-plot-all-scaling-py

**Comment by sofianehaddad** [comment]
@mbaudin47 It will be nice to have several features (in general). Note that the context in `scikit` is a bit different. The usage of these features is eased by the [pipeline](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html). You can define the full pipeline and do not have to handle manually the mapping in your process.

Here it is more tricky and depend on the application/context. If we focus only on `kriging` and its applications, evaluating surrogate/conditional variance require the transformation in the loop. No complexity here.
But let us suppose a more complex situation. For example you are performing `EGO`. The `kriging result` is obtained using the normalised data. That is a point. To go deeper, you need to define the optimisation problem, at least `objective` function and `bounds`. Here bounds should be accurate with respect to the `standard space`, ie the space corresponding to the transformed (normalised) data. The objective should be a `ComposedFunction` (`model` o `inverseTransformation`) if model takes into account variables in the original space. It is a bit complex in comparison with the first use-case

If we consider other situation (`SimulationAlgorithm` for example), the situation also differs. We need to transform the generated realization, redefine the event...

In other words, you see that complexity is not the definition of the transformation but how we should propagate it inside the algorithms. Implementation of transformations might be helpful but the real difficulties are elsewhere !

**Comment by mbaudin47** [comment]
What if the g function is pre-processed, so that a new function, operating on a standard space, is created before using the algorithm? Suppose that t is the transformation which maps from the physical space to the standard space. Pre-processing could be done with a `ComposedFunction` where the first function is g and the second is the inverse transformation. This way, the kriging could be safely performed on the composed function and EGO would run fine, isn'it?

**Comment by sofianehaddad** [comment]
If all elements are well defined it will be ok. The point is that it is more tricky to assess this part than the definition of the normalisation.
Users should be aware of how to define the full pipeline ....
